src_c/event.c

```diff
@@ -1751,20 +1753,75 @@ get_grab(PyObject *self)
 #endif /* IS_SDLv2 */
 }
 
+static void
+_pg_event_pump(int dopump)
+{
+    if (dopump) {
+        SDL_PumpEvents();
+    }
+#if IS_SDLv2
+    /* We need to translate WINDOWEVENTS. But if we do that from the
+     * from event filter, internal SDL stuff that rely on WINDOWEVENT
+     * might break. So after every event pump, we translate events from
+     * here */
+    SDL_FilterEvents(_pg_translate_windowevent, NULL);
+#endif
+}
+
+static int
+_pg_event_wait(SDL_Event *event, int timeout)
+{
+    /* Custom re-implementation of SDL_WaitEventTimeout, doing this has
```

Comment:
Can we write here what the advantages are? I guess it mainly uses our pump/peek implementations?

Reply:
Yes. We have our own pump function, that we should be using, going forward. This is to trigger the SDL_WINDOWEVENT translation